padd.sh

```diff
@@ -1387,6 +1387,58 @@ NormalPADD() {
   done
 }
 
+Update() {
+    GetPADDInformation
+
+    if [ "${padd_out_of_date_flag}" = "true" ]; then
+        echo "${full_status_update}"
+
+        padd_script_path="$0"
+        padd_script_file_name="padd.sh"
+        padd_script_directory="$(pwd -P)"
+
+        # Resolve potential symlinks until the script path is found
+        while :; do
+            if [ ! -L "${padd_script_path}" ] && [ ! -e "${padd_script_path}" ]; then
+                echo "${check_box_bad} Cannot locate script to update"
+                exit 1
+            fi
+            if ! cd "$(dirname -- "${padd_script_path}")"; then
+                echo "${check_box_bad} Cannot locate script to update"
+                exit 1
+            fi
+            padd_script_file_name="$(basename -- "${padd_script_path}")"
+            [ "${padd_script_file_name}" = '/' ] && padd_script_file_name=''
+            if [ -L "${padd_script_file_name}" ]; then
+                padd_script_path="$(ls -l "${pad_script_file_name}")"
+                padd_script_path="${padd_script_path#* -> }"
+                continue
+            fi
+            break
+        done
+
+        padd_script_path="${padd_script_directory%/}"/"${padd_script_file_name}"
```

Comment:
Why don't you use this?

```suggestion
        padd_script_path=$(realpath "$0")
```

Reply:
I'd love to do that and it would be way easier. However `realpath` does not seem to be part of the POSIX standard (https://stackoverflow.com/a/11114547 / https://pubs.opengroup.org/onlinepubs/9699919799/idx/utilities.html).
Not sure if this is a deal-breaker here 🤷🏼

Reply:
We don't install `coreutils` explicitly, but we rely on many of its tools even during the Pi-hole install process (`install`, `cp`, `mkdir`,..). Coreutils v8.15 (first containing `realpath`) was released >10 years ago, so I think it's safe to assume a recent version is installed on most linux systems today. And it should be installed on any currently Pi-hole supported OS.